### PR TITLE
UI: fix UI block after dragging source #2609

### LINKED ISSUE #2609: [UI Stuck] UI will be frozen while dragging source to move its order
Operating system and version: WIN10 64b
OBS Studio version: OBS.25.0.1
reproduced 100%

## Expected Behavior
Please pay attention that this issue cann't be reproduced except my special operation as below.
Steps:
1. add a group and some other sources, move group source to top.
2. resize height of source dock to ensure vertical scroll is visible.
3. drag the bottom source and move mouse to top position of source tree.
4. keep position of mouse on top of source tree, then vertical scroll will be moved automatically to the top  (don't move mouse while scroll is moving!).
5. mouse button up to end dragging, then the dragged source will be moved to the first one (incorrect index evidently, it should be moved into group)
6. try to drag the first source into group, then UI will be frozen.

## Reason
It is caused by logic about calculating row in SourceTree::dropEvent :

**It seems QAbstractItemView::OnViewport should also be considered.**
In step (4), the type is OnViewport and order_list is wrong which will cause loop circle for item list.

Record image:

### PR DESCRIPTION
**Pull request by wangshaohui1314**
### Description

UI: fix UI block after dragging source #2609

### Motivation and Context

drag source as special steps, UI will block. Fixes #2609.

### How Has This Been Tested?

win10 works fine

### Types of changes

Bug fix (non-breaking change which fixes an issue) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -1892,7 +1892,7 @@ void OBSBasic::OBSInit()
 	} else if (isWayland) {
 		if (opt_always_on_top)
 			blog(LOG_INFO,
-			     "Always On Top not available on Wayland, ignoring…");
+			     "Always On Top not available on Wayland, ignoring...");
```
Comment (dodgepong) [inline UI/window-basic-main.cpp:1895]: How was this line causing a compile error?

Also, this commit is unrelated to this PR and probably should have been in a separate PR.

Comment (wangshaohui1314) [reply UI/window-basic-main.cpp:1895]: > How was this line causing a compile error?
> 
> Also, this commit is unrelated to this PR and probably should have been in a separate PR.

The old character '…' causes a compile error on my visual studio 2019. So I replace it with three characters ‘.’

**Comment by wangshaohui1314** [comment]
Any other friends meet this compile error on vs2019 ?

**Comment by RytoEX** [comment]
> Any other friends meet this compile error on vs2019 ?
> Error can be fixed by replacing '…' with three '.'
> 
> 

I compile on Windows with VS2019 without this error.  It must be an issue with Windows or VS2019 and specific language settings.  The error message is strange since the file is already UTF-8.  As @dodgepong said, this change should be a separate PR.

**Comment by wangshaohui1314** [comment]
> > Any other friends meet this compile error on vs2019 ?
> > Error can be fixed by replacing '…' with three '.'
> > 
> 
> I compile on Windows with VS2019 without this error. It must be an issue with Windows or VS2019 and specific language settings. The error message is strange since the file is already UTF-8. As @dodgepong said, this change should be a separate PR.

I have reset the second commit, new pull request will be created later.

**Comment by RytoEX** [comment]
> > > Any other friends meet this compile error on vs2019 ?
> > > Error can be fixed by replacing '…' with three '.'
> > 
> > 
> > I compile on Windows with VS2019 without this error. It must be an issue with Windows or VS2019 and specific language settings. The error message is strange since the file is already UTF-8. As @dodgepong said, this change should be a separate PR.
> 
> I have reset the second commit, new pull request will be created later.
> @RytoEX To modify this compile error, should I also create a issue firstly ? Or commit a PR directly without issue ?

Sorry, I did not see the question.

> Or commit a PR directly without issue ?

You can submit a PR directly for the issue without creating an issue.

**Comment by wangshaohui1314** [comment]
That special character has been removed in another PR https://github.com/obsproject/obs-studio/pull/4420

**Pull request merged**